Source/Scene/UrlTemplateImageryProvider.js

```diff
@@ -274,6 +244,11 @@ define([
          */
         tileWidth : {
             get : function() {
+                //>>includeStart('debug', pragmas.debug);
+                if (!this.ready) {
+                    throw new DeveloperError('tileWidth must not be called before the imagery provider is ready.');
```

Comment:
Pass in a deferred (unresolved) promise as the options object and then ready will be false and you can test each property/method.